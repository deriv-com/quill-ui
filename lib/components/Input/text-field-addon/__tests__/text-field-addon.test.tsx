import { render } from "@testing-library/react";
import TextFieldAddon, { TextFieldAddonProps } from "..";
import React from "react";

describe('TextFieldAddon component', () => {
    const defaultProps: TextFieldAddonProps = {
        addonLabel: 'Test Label',
        addOnPosition: 'left',
        variant: 'outline',
        inputSize: 'md',
    };

    it('should render correctly with default props', () => {
        const { getByText, container } = render(<TextFieldAddon {...defaultProps} />);
        expect(getByText('Test Label')).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with the addon on the right', () => {
        const props: TextFieldAddonProps = { ...defaultProps, addOnPosition: 'right' };
        const { getByText, container } = render(<TextFieldAddon {...props} />);
        expect(getByText('Test Label')).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with a different variant', () => {
        const props: TextFieldAddonProps = { ...defaultProps, variant: 'fill' };
        const { getByText, container } = render(<TextFieldAddon {...props} />);
        expect(getByText('Test Label')).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with a right icon', () => {
        const props = { ...defaultProps, rightIcon: <span className="test-icon">Icon</span> };
        const { getByText, container } = render(<TextFieldAddon {...props} />);
        expect(getByText('Test Label')).toBeInTheDocument();
        expect(getByText('Icon')).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

})