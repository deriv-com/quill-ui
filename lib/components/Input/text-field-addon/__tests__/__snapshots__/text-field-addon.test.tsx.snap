// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldAddon component should render correctly with default props 1`] = `
<div
  class="quill-input__container"
>
  <div
    class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md"
  >
    <div
      class="quill-addon quill-addon__position--left quill-addon__position--left--outline quill-addon__variant--outline"
    >
      <p
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Test Label
      </p>
    </div>
    <div
      class="quill-input-icons__wrapper"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          id="mock-uuid"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <span
        class="icon_wrapper"
      />
    </div>
  </div>
</div>
`;

exports[`TextFieldAddon component should render with a different variant 1`] = `
<div
  class="quill-input__container"
>
  <div
    class="quill-input__wrapper quill-input__wrapper__variant--fill quill-input__wrapper__variant--fill--neutral quill-input__wrapper__size--md"
  >
    <div
      class="quill-addon quill-addon__position--left quill-addon__position--left--fill quill-addon__variant--fill"
    >
      <p
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Test Label
      </p>
    </div>
    <div
      class="quill-input-icons__wrapper"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          id="mock-uuid"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <span
        class="icon_wrapper"
      />
    </div>
  </div>
</div>
`;

exports[`TextFieldAddon component should render with a right icon 1`] = `
<div
  class="quill-input__container"
>
  <div
    class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md"
  >
    <div
      class="quill-addon quill-addon__position--left quill-addon__position--left--outline quill-addon__variant--outline"
    >
      <p
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Test Label
      </p>
    </div>
    <div
      class="quill-input-icons__wrapper"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          id="mock-uuid"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <span
        class="icon_wrapper"
      >
        <span
          class="test-icon"
        >
          Icon
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`TextFieldAddon component should render with the addon on the right 1`] = `
<div
  class="quill-input__container"
>
  <div
    class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md"
  >
    <div
      class="quill-addon quill-addon__position--right quill-addon__position--right--outline quill-addon__variant--outline"
    >
      <p
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Test Label
      </p>
    </div>
    <div
      class="quill-input-icons__wrapper"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          class="input peer input__align--left input__size--md"
          id="mock-uuid"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <span
        class="icon_wrapper"
      />
    </div>
  </div>
</div>
`;
